### PR TITLE
Bug: connectivity measures with LedoitWolf and wrong dimension

### LINKED ISSUE #941: ConnectivityMeasure requires 3D data
Otherwise it does strange stuff. Minimal example to come.

### PR DESCRIPTION
**Pull request by salma1601**
try to fix #941 and #955

### CODE REVIEW

File: nilearn/connectome/connectivity_matrices.py [hunk 1]
```diff
@@ -307,6 +308,26 @@ def fit(self, X, y=None):
             The object itself. Useful for chaining operations.
         """
         self.cov_estimator_ = clone(self.cov_estimator)
+        if not hasattr(X, "__iter__"):
+            raise ValueError("'subjects' input argument must be an iterable. "
+                             "You provided {0}".format(X.__class__))
+
+        subjects_types = [type(s)for s in X]
+        if set(subjects_types) != {np.ndarray}:
```
Comment (AlexandreAbraham) [inline nilearn/connectome/connectivity_matrices.py]: The syntax `{...}` is not valid in Python 2.6. Please use `set([...])`.

**Comment by AlexandreAbraham** [comment]
Your PR looks good to me. My comment should fix the Travis failure. Are you planning to do something else in it or can I merge it?

**Comment by salma1601** [comment]
I will only fix the Travis failure, thanks !

**Comment by salma1601** [comment]
may be I should remove the error raised for unknow kind from fit_transform to fit, what do you think ?

**Comment by AlexandreAbraham** [comment]
> may be I should remove the error raised for unknow kind from fit_transform to fit, what do you think ?

Let's move on.

**Pull request merged**